x-pack/legacy/plugins/ml/public/components/navigation_menu/tabs.tsx

```diff
@@ -71,14 +67,14 @@ function getTabs(disableLinks: boolean): Tab[] {
   ];
 }
 
-const TAB_TEST_SUBJ_MAP: TestSubjMap = {
-  jobs: 'mlTabJobManagement',
-  explorer: 'mlTabAnomalyExplorer',
-  timeseriesexplorer: 'mlTabSingleMetricViewer',
-  data_frames: 'mlTabDataFrames',
-  datavisualizer: 'mlTabDataVisualizer',
-  settings: 'mlTabSettings',
-};
+enum TAB_TEST_SUBJ_MAP {
```

Comment:
nit: this is more about `as keyof typeof`, I hadn't seen that before but it looks really useful! My suggestion is about just a bit of renaming to make it more obvious what it does.

- Since `TAB_TEST_SUBJ_MAP` is now an `enum` you could remove `_MAP` and name it `TAB_TEST_SUBJECT` (singular enum convention)
- An additional type then could be `const type TAB_TEST_SUBJECTS = keyof typeof TAB_TEST_SUBJECT` so later on in JSX you'd reference this instead of the inline definition.

If you agree to the change, I'll refactor to the same pattern some of the data frames code.

Reply:
Good one, @walterra! 👍 Updated in b48feda7ed3d0d8103c0c0fd861a4564ba51f876